operations.py

```diff
@@ -38,12 +64,341 @@ def kleene_star(automaton1: Automaton) -> Automaton:
 
 def get_states_cross_product(automaton1: Automaton, automaton2: Automaton) -> set:
     states = set()
+    join_set = automaton1.states | automaton2.states
     for s1 in sorted(automaton1.states):
-        for s2 in sorted(automaton2.states):
-            state = State(f"{s1.name}{s2.name}")
+        for s2 in sorted(automaton2.states):         
+            state = State(clean_target(f"{s1.name}{s2.name}", join_set))
             for trans1 in s1.transitions:
                 for trans2 in s2.transitions:
                     if trans1.symbol == trans2.symbol:
-                        state.add_transition(f"{trans1.target}{trans2.target}", trans1.symbol)
+                        ctarget = clean_target(f"{trans1.target}{trans2.target}", join_set)
+                        state.add_transition(ctarget, trans1.symbol)
             states.add(state)
     return states
+
+def is_afd(automaton: Automaton) -> bool:
+    for s1 in sorted(automaton.states):
+        if len(s1.transitions) != len(automaton.alphabet):
+            return False
+
+        for alph in automaton.alphabet:
+            found = False
+            for trans in s1.transitions:
+                if trans.symbol == EPSILON_SYMBOL:
+                    return False
+                elif trans.symbol == alph:
+                    found = True
+
+            if not found:
+                return False
+            
+    return True
```

Comment:
We can save a lot of iterations if we check `is_afd` and `has_epsilon` during validation and add them as attributes of `Automaton`